cmake/modules/SetUpMacOS.cmake

```diff
@@ -54,16 +54,24 @@ if (CMAKE_SYSTEM_NAME MATCHES Darwin)
     set(MACOSX_GLU_DEPRECATED ON)
   endif()
 
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15)
```

Comment:
Look at the semantic (see man ld) of both, I would try to remove them entirely.    Seemingly `--single_module` has been ignore since MacOS 10.4.  I am not sure why `-bind_at_load` is being removed/deprecated but we should be testing without it on all the MacOS version to understand where it works without it (likely all the current version of MacOS).